### DISCUSSION TITLE
Applying isOwner Policy to user-permissions Plugin

### DISCUSSION THREAD

**Discussion by spicy-sauce** [strapi/strapi#8970] [Legacy Topics]
Hi,
I'm trying to implement JIm's suggestion described here [1] in order to restrict users to update only their data/user profile (by implementing an isOwner policy).
So, I've created isOwner.js placed it in /config/policies:
module.exports = async (ctx, next) => {
  if (ctx.state.user.id === ctx.params.id) {
    // Go to next policy or will reach the controller's action.
    return await next();
  }

  ctx.unauthorized(`You're not allowed to perform this action!`);
};

Also created a routes.json file under /extensions/user-permissions/config with this route at this point:
{
  "routes": [
    {
      "method": "PUT",
      "path": "/users/:id",
      "handler": "User.update",
      "config": {
        "policies": [
          "global::isOwner"
        ]
      }
    }
  ]
}

My intention here was to override the plugin's built-in route so it will go through the new policy, but it seems not to be working as the old/other definition is used.
What am I doing wrong?

Best regards,
Yossi

[1] https://stackoverflow.com/questions/59645357/how-to-change-password-after-logging-in

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question, we are currently planning to decomission our GitHub Discussions in early 2021 and we don't want to lose the great questions asked by our community.

Questions should be directed to [our forum](https://forum.strapi.io/). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).
